satpy/demo/fci.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Demo FCI data download."""
+
+import urllib
+import tarfile
+import pathlib
+from satpy import config
+
+_fci_uncompressed_nominal = (
+    "https://sftp.eumetsat.int/public/folder/UsCVknVOOkSyCdgpMimJNQ/"
+    "User-Materials/Test-Data/MTG/MTG_FCI_L1C_Enhanced-NonN_TD-272_May2020/"
+    "FCI_1C_UNCOMPRESSED_NOMINAL.tar.gz")
+
+
+def download_fci_test_data(base_dir=None):
+    """Download FCI test data.
+
+    Download the nominal FCI test data from July 2020.
+    """
+    subdir = _get_fci_test_data_dir(base_dir=base_dir)
+    (local_filename, headers) = urllib.request.urlretrieve(_fci_uncompressed_nominal)
+    return _unpack_tarfile_to(local_filename, subdir)
+
+
+def _get_fci_test_data_dir(base_dir=None):
+    """Get directory for FCI test data."""
+    base_dir = base_dir or config.get("demo_data_dir", ".")
+    return pathlib.Path(base_dir) / "fci" / "test_data"
+
+
+def _unpack_tarfile_to(filename, subdir):
+    """Unpack content of tarfile in filename to subdir."""
+    with tarfile.open(filename, mode="r:gz") as tf:
+        contents = tf.getnames()
+        tf.extractall(path=subdir)
```

Comment:
I think deepcode says it doesn't like the use of `extractall`. Any thoughts/opinions on its suggestion?

Reply:
Deepcode's suggestion refers to a [9 year old Stack Overflow answer](https://stackoverflow.com/a/10077309/974555) that seems quite complicated.  Do we need this for a hardcoded EUMETSAT archive?  Getting this right is a non-trivial problem.

Reply:
Ah forget it then. I saw the first suggested diff which was basically doing what you are already doing (now that I look at it again). Ignore this.